gulp-tasks/utils/publish-helpers.js

```diff
@@ -0,0 +1,179 @@
+const fs = require('fs-extra');
+const path = require('path');
+const glob = require('glob');
+const archiver = require('archiver');
+const oneLine = require('common-tags').oneLine;
+
+const spawn = require('./spawn-promise-wrapper');
+const constants = require('./constants');
+const logHelper = require('../../infra/utils/log-helper');
+
+const SOURCE_CODE_DIR = 'source-code';
+const GROUPED_BUILD_FILES = 'grouped-build-files';
+
+const doesDirectoryExist = async (directoryPath) => {
+  let stats = null;
+  try {
+    stats = await fs.stat(directoryPath);
+  } catch (err) {
+    // NOOP
+  }
+  return stats !== null;
+};
+
+const getBuildPath = (tagName) => {
+  const tempReleasePath = path.join(
+    __dirname, '..', '..', constants.GENERATED_RELEASE_FILES_DIRNAME);
+  return path.join(tempReleasePath, tagName);
+};
+
+const downloadGitCommit = async (tagName, gitBranch) => {
+  if (!tagName) {
```

Comment:
Completely optional, but from Node code, I've just been using the built-in [`assert` module](https://nodejs.org/api/assert.html) for this sort of thing.